### PR TITLE
Enable thetis adjoint

### PR DESCRIPTION
**Pull request by tkarna**

### CODE REVIEW

**Comment by tkarna** [comment]
This looks all right to me as long as travis is happy.

File: test_adjoint/test_swe_adjoint.py [hunk 1]
```diff
@@ -93,6 +93,8 @@ def setup_unsteady():
     # automatically switches off matnest when using 'lu',
     # firedrake-adjoint does not remember that in the adjoint solve
     parameters['matnest'] = False
+    # coffee optimisation currently broken for this case:
+    parameters['coffee'] = {}
```
Comment (tkarna) [inline test_adjoint/test_swe_adjoint.py]: Ah, thanks for fixing these. I had no idea these were required for adjoint.

Comment (stephankramer) [inline test_adjoint/test_swe_adjoint.py]: No, neither did I. They're bugs really. One is fixed already by Fabio:
https://github.com/coneoproject/COFFEE/issues/81
and the other is a missing feature in firedrake-adjoint:
https://bitbucket.org/dolfin-adjoint/dolfin-adjoint/issues/43/firedrake-adjoint-does-not-respect-nest

Comment (tkarna) [inline test_adjoint/test_swe_adjoint.py]: I see, great. Should we merge this pr and deal with those later?

**Pull request merged**